autoscale/src/main/java/com/sequenceiq/periscope/service/ScalingActivityService.java

```diff
@@ -66,10 +67,28 @@ public List<Long> findAllIdsOlderThanDurationForCluster(Cluster cluster, long du
                 .stream().map(ScalingActivity::getId).collect(Collectors.toList());
     }
 
+    public List<ScalingActivity> findAllInGivenIntervalForCluster(Cluster cluster, long durationInMinutes, Pageable pageable) {
+        Date startTimeAfter = new Date(Instant.now().minus(durationInMinutes, ChronoUnit.MINUTES).toEpochMilli());
+        return scalingActivityRepository.findAllByClusterWithStartTimeAfter(cluster.getId(), startTimeAfter, pageable);
+    }
+
+    public List<ScalingActivity> findAllWithTriggerFailed(Cluster cluster, long durationInMinutes, Pageable pageable) {
+        Date startTimeAfter = new Date(Instant.now().minus(durationInMinutes, ChronoUnit.MINUTES).toEpochMilli());
+        return scalingActivityRepository.findAllByClusterAndInStatusesAndInGivenInterval(cluster.getId(),
+                newArrayList(ActivityStatus.DOWNSCALE_TRIGGER_FAILED, ActivityStatus.UPSCALE_TRIGGER_FAILED,
+                        ActivityStatus.METRICS_COLLECTION_FAILED), startTimeAfter, pageable);
+    }
+
     public List<ScalingActivity> findAllScalingActivityWithStatusesForCluster(Cluster cluster, Collection<ActivityStatus> statuses) {
         return scalingActivityRepository.findAllByClusterAndInStatuses(cluster.getId(), statuses);
     }
 
+    public List<ScalingActivity> findAllInTimeRangeForCluster(Cluster cluster, long timestampFrom, long timestampUntil, Pageable pageable) {
+        Date startTimeFrom = new Date(timestampFrom);
+        Date startTimeUntil = new Date(timestampUntil);
+        return scalingActivityRepository.findAllByClusterBetweenInterval(cluster.getId(), startTimeFrom, startTimeUntil, pageable);
+    }
```

Comment:
The default constructor for `Date` takes the epoch value in milliseconds. I don't see this unit specified anywhere in the API call chain, which can cause ambiguity. Please specify a particular unit, possibly `minutes`, as part of the argumentName - similar to the coolDownMinutes param when creating an autoscaling policy. This should be consistent throughout the API call chain